main.py

```diff
@@ -1,24 +1,30 @@
 #!/bin/python3
 
-from socket import AF_INET, SOCK_STREAM, socket as os_socket
+from socket import AF_INET, SOCK_STREAM, gethostname, gethostbyname, socket as os_socket
 import sys
 import json
 import queue
 import logging
 from threading import Thread
 
 APPLICATION_PORT = 65412
+startup = None
 logger = logging.getLogger(__name__)
 logging.basicConfig(filename=f"chat.log", level=logging.DEBUG, format="%(asctime)s - %(message)s")
 
 class Node:
     def __init__(self, hosts, nickname):
         self.message_queue = queue.Queue()
-        self.peer_hosts = hosts
+        self.peer_hosts = hosts if hosts is not None else []
```

Comment:
I would store the peer_hosts in set so you don't need to constantly convert it from and to set.